twake/frontend/src/app/services/Apps/Messages/Messages.js

```diff
@@ -142,6 +142,59 @@ class Messages extends Observable {
       var message = Collections.get('messages').edit();
       var val = PseudoMarkdownCompiler.compileToJSON(value);
 
+      const editorManager = MessageEditorsManager.get(options.channel_id);
+      let filesAttachements =
+        editorManager.filesAttachements[options.parent_message_id || 'main'] || [];
+
+      const filesAttachementsToTwacode =
+        filesAttachements.map(id => {
+          return {
+            type: 'file',
+            mode: filesAttachements.length > 1 ? 'mini' : 'preview',
+            content: id,
+          };
+        }) || {};
+
+      const user = UserService.getCurrentUser();
+
+      const singleFileSystemMessage =
+        val.original_str.length === 0
+          ? [
+              { type: 'br' },
+              {
+                type: 'system',
+                content: Languages.t('scenes.apps.drive.message_added_file_no_name', [
+                  UserService.getFullName(user),
+                ]),
+              },
+              { type: 'br' },
+            ]
+          : [{ type: 'br' }];
+
+      const multipleFileSystemMessage =
+        val.original_str.length === 0
+          ? [
+              { type: 'br' },
+              {
+                type: 'system',
+                content: Languages.t('scenes.apps.drive.message_added_mutiple_files', [
+                  UserService.getFullName(user),
+                ]),
+              },
+              { type: 'br' },
+            ]
+          : [{ type: 'br' }];
+
+      const preparedFiles = filesAttachementsToTwacode.length
+        ? [
+            filesAttachements.length >= 2
+              ? [...multipleFileSystemMessage]
+              : [...singleFileSystemMessage],
+            ...filesAttachementsToTwacode,
+          ]
+        : [];
+
```

Comment:
This is a lot of code, can we move it to a function ?

Reply:
Done :+1: